custom/icds_reports/static/js/directives/month-filter/month-filter.directive.js

```diff
@@ -28,30 +45,47 @@ function MonthModalController($location, $uibModalInstance, dateHelperService) {
     vm.selectedMonth = dateHelperService.getSelectedMonth();
     vm.selectedYear = dateHelperService.getSelectedYear();
 
+
+    var fullselectedDate = new Date(vm.selectedYear, vm.selectedMonth - 1);
+
     if (isSDD && (vm.selectedYear < 2019 || (vm.selectedYear === 2019 && vm.selectedMonth === 1))) {
         vm.showMessage = true;
         vm.selectedYear = new Date().getFullYear();
         vm.selectedMonth = new Date().getMonth() + 1;
     }
 
+    if (haveAccessToFeatures && isAdolescentGirls && fullselectedDate < reportStartDates['adolescent_girls']) {
+        vm.showAdolescentGirlMessage = true;
+        vm.selectedYear = new Date().getFullYear();
+        vm.selectedMonth = new Date().getMonth() + 1;
+    }
+
+    vm.months = vm.monthsCopy;
+
     if (vm.selectedYear === new Date().getFullYear()) {
         vm.months = _.filter(vm.monthsCopy, function (month) {
             return month.id <= new Date().getMonth() + 1;
         });
 
-        if (startDate === 2019) {
-            vm.months.shift();
-        }
-    } else if (startDate === 2019 && vm.selectedYear === 2019) {
-        vm.months = _.filter(vm.monthsCopy, function (month) {
-            return month.id >= 2;
-        });
-    } else if (vm.selectedYear === 2017) {
-        vm.months = _.filter(vm.monthsCopy, function (month) {
-            return month.id >= 3;
-        });
     } else {
-        vm.months = vm.monthsCopy;
+
+        if (isSDD && vm.selectedYear === reportStartDates['sdd'].getFullYear()) {
```

Comment:
could reduce brain overhead if each of these clauses had a short comment explaining what they are doing. I realize it's relatively clear once you get into it, but the complex conditionals are a bit unwieldy and it relies on knowing, e.g. dashboard launched on march 2017.